src/index.js

```diff
@@ -63,54 +63,51 @@ export function manipulatePluginOptions(pluginOpts) {
 }
 
-export default ({ types: t }) => ({
-  manipulateOptions(babelOptions) {
-    let findPluginOptions = babelOptions.plugins.find(plugin => plugin[0] === this)[1];
-    findPluginOptions = manipulatePluginOptions(findPluginOptions);
-
-    this.customCWD = findPluginOptions.cwd;
-  },
-
-  pre(file) {
-    let { customCWD } = this.plugin;
-    if (customCWD === 'babelrc') {
-      const startPath = (file.opts.filename === 'unknown')
-        ? './'
-        : file.opts.filename;
-
-      const { file: babelFile } = findBabelConfig.sync(startPath);
-      customCWD = babelFile
-        ? path.dirname(babelFile)
-        : null;
-    }
-
-    this.moduleResolverCWD = customCWD || process.cwd();
-  },
-
-  visitor: {
-    CallExpression: {
-      exit(nodePath, state) {
-        if (nodePath.node.seen) {
-          return;
-        }
+export default ({ types: t }) => {
+  const importVisitors = {
+    CallExpression(nodePath, state) {
+      transformRequireCall(t, nodePath, mapModule, state, this.moduleResolverCWD);
+      transformJestCalls(t, nodePath, mapModule, state, this.moduleResolverCWD);
+      transformSystemImportCall(t, nodePath, mapModule, state, this.moduleResolverCWD);
+    },
+    ImportDeclaration(nodePath, state) {
+      transformImportCall(t, nodePath, mapModule, state, this.moduleResolverCWD);
+    },
+    ExportDeclaration(nodePath, state) {
+      transformImportCall(t, nodePath, mapModule, state, this.moduleResolverCWD);
+    },
+  };
 
-        transformRequireCall(t, nodePath, mapModule, state, this.moduleResolverCWD);
-        transformJestCalls(t, nodePath, mapModule, state, this.moduleResolverCWD);
-        transformSystemImportCall(t, nodePath, mapModule, state, this.moduleResolverCWD);
+  return {
+    manipulateOptions(babelOptions) {
+      let findPluginOptions = babelOptions.plugins.find(plugin => plugin[0] === this)[1];
+      findPluginOptions = manipulatePluginOptions(findPluginOptions);
 
-        // eslint-disable-next-line no-param-reassign
-        nodePath.node.seen = true;
-      },
+      this.customCWD = findPluginOptions.cwd;
     },
-    ImportDeclaration: {
-      exit(nodePath, state) {
-        transformImportCall(t, nodePath, mapModule, state, this.moduleResolverCWD);
-      },
+
+    pre(file) {
+      let { customCWD } = this.plugin;
+      if (customCWD === 'babelrc') {
+        const startPath = (file.opts.filename === 'unknown')
+          ? './'
+          : file.opts.filename;
+
+        const { file: babelFile } = findBabelConfig.sync(startPath);
+        customCWD = babelFile
+          ? path.dirname(babelFile)
+          : null;
+      }
+
+      this.moduleResolverCWD = customCWD || process.cwd();
     },
-    ExportDeclaration: {
-      exit(nodePath, state) {
-        transformImportCall(t, nodePath, mapModule, state, this.moduleResolverCWD);
+
+    visitor: {
+      Program: {
+        exit(programPath, state) {
+          programPath.traverse(importVisitors, state);
```

Comment:
How good/bad this is?

Is it the same in term of performance as what was done before? I remember reading that if we can do something without `traversing`, it's usually best